examples/compiler/01_iet.ipynb

```diff
@@ -0,0 +1,664 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This tutorial describes one particular intermediate representation used by the Devito Compiler: the Iteration/Expression Tree (IET), a special type of Abstract Syntax Tree (AST).\n",
+    "\n",
+    "# Part I - Top Down\n",
+    "\n",
+    "First, let's describe a _domain_, and a _function_ that will allow us to specify how such a domain gets modified. In particular, we will look at Functions that change through _time_. \n",
+    "\n",
+    "Thus we need a `Function` object with which we can build a timestepping scheme. For this purpose Devito provides so-called `TimeFunction` objects that encapsulate Functions that are differentiable in space and time, which are derived from basic _SymPy_ Functions. \n",
+    "\n",
+    "With this we can derive symbolic expressions for the backward derivatives in space directly via the `u.dxl` and `u.dyl` shorthand expressions (the l indicates \"left\" or backward differences) and the shorthand notation `u.dt` provided by `TimeFunction` objects to derive the forward derivative in time.\n",
```

Comment:
`u.dl` is not backward derivative, it is a derivative centered left of the current point, you can see it for high orders and you'll see it is not. `backward/forward` or `upwind/downwind` derivatives don't have shortcuts yet, namings needs some fix too.